packages/react-dom/src/shared/ensureCorrectIsomorphicReactVersion.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import reactDOMPackageVersion from 'shared/ReactVersion';
+import * as IsomorphicReactPackage from 'react';
+
+export function ensureCorrectIsomorphicReactVersion() {
+  const isomorphicReactPackageVersion = IsomorphicReactPackage.version;
+  if (isomorphicReactPackageVersion !== reactDOMPackageVersion) {
+    // TODO: Do we have a canoncical documentation page for this?
```

Comment:
Closest is https://legacy.reactjs.org/warnings/invalid-hook-call-warning.html#mismatching-versions-of-react-and-react-dom and https://react.dev/warnings/invalid-hook-call-warning#mismatching-versions-of-react-and-react-dom. But these are just about a React version that's too low or duplicate React modules not explicitly about version mismatch. For example, package managers that auto-install peer dependencies will result in multiple versions of React if the specified version of `react` and `react-dom` don't match.

So maybe we include version mismatch in that section and link to it?

Reply:
Just added `react.dev/warnings/version-mismatch`, @acdlite can we add that URL to the error?

Reply:
It just redirects for now, but I'll follow up with a proper warning page